tests/drivers/sensor/f75303/src/main.c

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2023 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/emul.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/ztest.h>
+#include <zephyr/drivers/sensor/f75303.h>
+#include "f75303.h"
+#include "f75303_emul.h"
+
+struct f75303_fixture {
+	const struct device *dev;
+	const struct emul *target;
+};
+
+static void *f75303_setup(void)
+{
+	static struct f75303_fixture fixture = {
+		.dev = DEVICE_DT_GET(DT_NODELABEL(f75303)),
+		.target = EMUL_DT_GET(DT_NODELABEL(f75303)),
+	};
+
+	zassert_not_null(fixture.dev);
+	zassert_not_null(fixture.target);
+	return &fixture;
+}
+
+static void f75303_before(void *f)
+{
+	struct f75303_fixture *fixture = f;
+
+	f75303_emul_reset(fixture->target);
+}
+
+ZTEST_SUITE(f75303, NULL, f75303_setup, f75303_before, NULL, NULL);
+
+static void check_val(struct sensor_value *val, uint16_t temp, const char *name)
+{
+	struct sensor_value expected;
+
+	expected.val1 = temp >> 3;
+	expected.val2 = (temp & 7) * 125000;
+
+	zassert_true(val->val1 == expected.val1 && val->val2 == expected.val2,
+		     "%s: expected val1=%d val2=%d, got val1=%d val2=%d",
+		     name, expected.val1, expected.val2, val->val1, val->val2);
+}
+
+static void test_fetch_temp(const struct f75303_fixture *fixture, uint16_t temp_local,
+			    uint16_t temp_remote1, uint16_t temp_remote2)
+{
+	struct sensor_value val_local;
+	struct sensor_value val_remote1;
+	struct sensor_value val_remote2;
+
+	f75303_emul_set_reg(fixture->target, F75303_LOCAL_TEMP_H, temp_local >> 3);
+	f75303_emul_set_reg(fixture->target, F75303_LOCAL_TEMP_L, (temp_local & 7) << 5);
```

Comment:
Thanks for creating an emulator and test!  However instead of having the test write register values directly into the emulator, you can use the [emulator sensor backend API](https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/drivers/emul_sensor.h).

The [generic sensor test](https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/drivers/build_all/sensor/src/generic_test.c) will then validate your new temperature sensor an emulator.

Your emulator code needs to implement the 'set_channel` API, which will internally write the correct registers corresponding to the set operation.

See the [akm09918c emulator](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/sensor/akm09918c/akm09918c_emul.c) for an example.

Reply:
Done